### PR TITLE
Add more indirect modes

### PR DESCRIPTION
**Pull request by BobVarioa**
Based on the discussion in the Porffor Discord, doesn't change any defaults, but `--indirect-call-mode=padding` maybe should be considered as one.

Examples
```
$ ./porf run ./bench/indirect/single.js --indirect-call-mode=padding
single arg indirect 48.30ms
$ ./porf run ./bench/indirect/single.js
single arg indirect 60.90ms
$ ./porf run ./bench/indirect/many.js --indirect-call-mode=padding
many arg indirect 52.40ms
$ ./porf run ./bench/indirect/many.js
many arg indirect 66.72ms

$ ./porf run ./bench/indirect/single.js --indirect-call-mode=padding --no-run -b
wasm size: 78354 bytes
$ ./porf run ./bench/indirect/single.js --no-run -b
wasm size: 79111 bytes
$ ./porf run ./bench/indirect/many.js --indirect-call-mode=padding --no-run -b
wasm size: 78336 bytes
$ ./porf run ./bench/indirect/many.js --no-run -b
wasm size: 79077 bytes

$ ./porf run ./bench/indirect/single.js
single arg indirect 59.99ms
$ ./porf run ./bench/indirect/single.js --indirect-call-mode=array
single arg indirect 64.82ms
$ ./porf run ./bench/indirect/many.js          
many arg indirect 66.05ms
$ ./porf run ./bench/indirect/many.js --indirect-call-mode=array
many arg indirect 93.20ms

$ ./porf run ./bench/indirect/single.js --no-run -b
wasm size: 79111 bytes
$ ./porf run ./bench/indirect/single.js --indirect-call-mode=array --no-run -b  
wasm size: 78506 bytes
$ ./porf run ./bench/indirect/many.js --no-run -b           
wasm size: 79077 bytes
$ ./porf run ./bench/indirect/many.js --indirect-call-mode=array --no-run -b 
wasm size: 78640 bytes
```

### CODE REVIEW

File: compiler/codegen.js [hunk 1]
```diff
@@ -56,6 +56,101 @@ const funcRef = func => {
     [ Opcodes.const, 'funcref', func.name ]
   ];
 
+  if (Prefs.indirectCallMode === 'array' && !func.internal) {
```
Comment (CanadaHonk) [inline compiler/codegen.js:59]: For now could you just add the `padding` mode to try and simplify internals for now? ty

**Review by CanadaHonk — CHANGES_REQUESTED** [review]

File: compiler/codegen.js (hunk 1 above)
Comment (BobVarioa) [reply compiler/codegen.js:59]: Just remove `array` right? Not switch `padding` to the default?

Comment (CanadaHonk) [reply compiler/codegen.js:59]: Yeah please

File: compiler/codegen.js [hunk 2]
```diff
@@ -2183,7 +2234,90 @@ const generateCall = (scope, decl, _global, _name, unusedValue = false) => {
       let args = decl.arguments;
       let locals = [];
 
-      if (indirectMode === 'vararg') {
+      if (indirectMode === 'array') {
```
Comment (CanadaHonk) [inline compiler/codegen.js:2237]: Forgot to remove this too? :sweat_smile:

**Review by CanadaHonk — CHANGES_REQUESTED** [review]

File: compiler/codegen.js (hunk 2 above)
Comment (BobVarioa) [reply compiler/codegen.js]: Woops :sweat_smile:

**Review by CanadaHonk — APPROVED** [review]
Looks good, thanks!

**Pull request merged**